src/devices/Bmm150/Bmm150.cs

```diff
@@ -0,0 +1,319 @@
+        public byte GetDeviceInfo() => ReadByte(Bmp180Register.INFO);
+
+        /// <summary>
+        /// Calibrate the magnetometer.
+        /// Please make sure you are not close to any magnetic field like magnet or phone
+        /// Please make sure you are moving the magnetometer all over space, rotating it.
+        /// </summary>
+        /// <param name="numberOfMeasurements">Number of measurement for the calibration, default is 100</param>
+        // https://platformio.org/lib/show/12697/M5_BMM150
+        public void CalibrateMagnetometer(int numberOfMeasurements = 100)
+        {
+            Vector3 mag_min = new Vector3() { X = 9000, Y = 9000, Z = 30000 };
+            Vector3 mag_max = new Vector3() { X = -9000, Y = -9000, Z = -30000 };
+            Vector3 rawMagnetometerData;
+
+            for (int i = 0; i < numberOfMeasurements; i++)
+            {
+                try
+                {
+                    rawMagnetometerData = ReadMagnetometerWithoutCorrection();
+
+                    if (rawMagnetometerData.X != 0)
+                    {
+                        mag_min.X = (rawMagnetometerData.X < mag_min.X) ? rawMagnetometerData.X : mag_min.X;
+                        mag_max.X = (rawMagnetometerData.X > mag_max.X) ? rawMagnetometerData.X : mag_max.X;
+                    }
+
+                    if (rawMagnetometerData.Y != 0)
+                    {
+                        mag_max.Y = (rawMagnetometerData.Y > mag_max.Y) ? rawMagnetometerData.Y : mag_max.Y;
+                        mag_min.Y = (rawMagnetometerData.Y < mag_min.Y) ? rawMagnetometerData.Y : mag_min.Y;
+                    }
+
+                    if (rawMagnetometerData.Z != 0)
+                    {
+                        mag_min.Z = (rawMagnetometerData.Z < mag_min.Z) ? rawMagnetometerData.Z : mag_min.Z;
+                        mag_max.Z = (rawMagnetometerData.Z > mag_max.Z) ? rawMagnetometerData.Z : mag_max.Z;
+                    }
+
+                    // Wait for 100ms until next reading
+                    Wait(100);
+                }
+                catch
+                {
+                    // skip this reading
+                }
+            }
+
+            // Refresh CalibrationCompensation vector
+            CalibrationCompensation = new Vector3()
+            {
+                X = (mag_max.X + mag_min.X) / 2,
+                Y = (mag_max.Y + mag_min.Y) / 2,
+                Z = (mag_max.Z + mag_min.Z) / 2
+            };
+        }
+
+        /// <summary>
+        /// True if there is a data to read
+        /// </summary>
+        public bool HasDataToRead => (ReadByte(Bmp180Register.DATA_READY_STATUS) & 0x01) == 0x01;
+
+        /// <summary>
+        /// Check if the version is the correct one (0x32). This is fixed for this device
+        /// </summary>
+        /// <returns>Returns true if the version match</returns>
+        public bool IsVersionCorrect => ReadByte(Bmp180Register.WIA) == 0x32;

```

Comment:
Unless the exception in the `Initialize` function can be prevented somehow, this never returns false for an outside caller. So maybe make private?